lib/wallaby/phantom/driver.ex

```diff
@@ -333,6 +333,135 @@ defmodule Wallaby.Phantom.Driver do
     end
   end
 
+  @doc """
+  Accept all JavaScript dialogs
+  """
+  def accept_dialogs(session) do
+    script = """
+    var page = this;
+    page.onAlert = function(msg) {}
+    page.onConfirm = function(msg) { return true; }
+    page.onPrompt = function(msg, defaultVal) { return defaultVal; }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+  end
+
+  @doc """
+  Dismiss all JavaScript dialogs
+  """
+  def dismiss_dialogs(session) do
+    script = """
+    var page = this;
+    page.onAlert = function(msg) {}
+    page.onConfirm = function(msg) { return false; }
+    page.onPrompt = function(msg, defaultVal) { return null; }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+  end
+
+  @doc """
+  Accept one alert triggered within `fun` and return the alert message.
+  """
+  def accept_alert(%Session{}=session, fun) do
+    script = """
+    var page = this;
+    page.__onAlertDefault = page.onAlert;
+    page.onAlert = function(msg) {
+      page.__alertMessage = msg;
+      page.onAlert = page.__onAlertDefault;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+    fun.(session)
+    script = """
+    var page = this;
+    page.onAlert = page.__onAlertDefault;
+    return ["ok", page.__alertMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
+  end
+
+  @doc """
+  Accept one confirm triggered within `fun` and return the confirm message.
+  """
+  def accept_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, true)
+  end
+
+  @doc """
+  Dismiss one confirm triggered within `fun` and return the confirm message.
+  """
+  def dismiss_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, false)
+  end
+
+  defp handle_confirm(%Session{}=session, fun, return_value) do
+    script = """
+    var page = this, returnVal = arguments[0];
+    page.__onConfirmDefault = page.onConfirm;
+    page.onConfirm = function(msg) {
+      page.__confirmMessage = msg;
+      page.onConfirm = page.__onConfirmDefault;
+      return returnVal;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script, [return_value])
+    fun.(session)
```

Comment:
- [ ] When invoking the callbacks like here, do we need to catch errors to make sure that the callback is properly reset? IMO this would only be a problem if:

* inside the callback an error is raised *before* the dialog is triggered
* the test that actually uses this `accept_*`/`dismiss_*` function catches errors itself
* so when the error was catched in the surrounding test, the callback handler is in place, but was not reset back to the default handler

Might be an edge case that's not worth handling...